### ISSUE TITLE
Feature request: Allowing custom ports

### ISSUE THREAD

**Issue by leonvendrame** [OneLiberty/moonlight-chrome-tizen#27]
Hey guys!

I want do request this freature if its possible.

I run my server on a custom port, because I have other pc already running on default ports.
On moonlight for android for an example, I can input my address like "serveraddress:port" and it runs perfectly.

If its way too trouble to do it, i'm sorry about the request.

**Comment by OneLiberty** [comment]
Can you send me your port configuration ? 
Is it for local or remote ?

**Comment by leonvendrame** [comment]
![image](https://github.com/OneLiberty/moonlight-chrome-tizen/assets/26984336/9f41d2e3-f0e7-4423-b097-1d490b447511)

I'm using for local and remote.

**Comment by OneLiberty** [comment]
So i did check a bit, and i must say, i have absolutely no idea on how to handle this :/ 
i'll keep looking tho. 

i'll let the issue open, maybe someone will have an idea.

**Comment by henryfa2** [comment]
Well, I would say we need : 

1. an interface to allow the user to parametrize the ports he wanna use,
2. an update to the code to change the ports : search in the code the different default values (for ex. 47984), analyze the code and replace the default values with the parametrized ones where it is needed

**Comment by Wacheee** [comment]
Yes please, I have two sunshine instances on my pc (thanks to Duo project) and two samsung TV. I cant connect to the instance that has a different port than 47989

**Comment by leonvendrame** [comment]
> Yes please, I have two sunshine instances on my pc (thanks to Duo project) and two samsung TV. I cant connect to the instance that has a different port than 47989

So, I managed to bypass this limitation in windows.
I'm using this to use it in my local network. 

My pc is running in another ports, but I'm using netsh commands to reroute the default sunshine ports to my custom ports.

netsh interface portproxy add v4tov4 listenport=45990 listenaddress=0.0.0.0 connectport=9000 connectaddress=127.0.0.1

netsh interface portproxy show all

![image](https://github.com/user-attachments/assets/153e2ea4-1326-4d26-9136-f5dd0915c948)

I don't know how to explain it better, but with a little bit of google you can find it.

It won't work in external networks because the exposed port will be to one of the pcs, but in the local network where you can point to a specific ip, it will be like your ports are working as default.

**Comment by Wacheee** [comment]
> > Yes please, I have two sunshine instances on my pc (thanks to Duo project) and two samsung TV. I cant connect to the instance that has a different port than 47989
> 
> So, I managed to bypass this limitation in windows. I'm using this to use it in my local network.
> 
> My pc is running in another ports, but I'm using netsh commands to reroute the default sunshine ports to my custom ports.
> 
> netsh interface portproxy add v4tov4 listenport=45990 listenaddress=0.0.0.0 connectport=9000 connectaddress=127.0.0.1
> 
> netsh interface portproxy show all
> 
> ![image](https://private-user-images.githubusercontent.com/26984336/382425113-153e2ea4-1326-4d26-9136-f5dd0915c948.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1MDU4ODgsIm5iZiI6MTczMDUwNTU4OCwicGF0aCI6Ii8yNjk4NDMzNi8zODI0MjUxMTMtMTUzZTJlYTQtMTMyNi00ZDI2LTkxMzYtZjVkZDA5MTVjOTQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAxVDIzNTk0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4MWI0ZGRmOTAxODkzNzkyMjVlM2M3YmJiMmMzMTMyOWViNzM3YWE4OWE2OTFmNjBiNWY5MDgyYzhjZTExOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x0UKwj6F6somrZiW8mqRqSJ9KF_B090gvEGdy6AkSeI)
> 
> I don't know how to explain it better, but with a little bit of google you can find it.
> 
> It won't work in external networks because the exposed port will be to one of the pcs, but in the local network where you can point to a specific ip, it will be like your ports are working as default.

In my case this didnt work because i need one TV to connect to a port (eg 47989) and in my second TV i need to connect to another port (eg 9000)

**Comment by eduardomozart** [comment]
> So i did check a bit, and i must say, i have absolutely no idea on how to handle this :/ i'll keep looking tho.
> 
> i'll let the issue open, maybe someone will have an idea.

Here's an in-depth documentation on how Sunshine calculates custom ports: https://deepwiki.com/LizardByte/Sunshine/4.5-network-configuration-and-port-management

It could be implemented when the user checks the box to type in the IP address, so it would be informed in the format: ``<IP>:<Port>``
